test/unit/classification_test.rb

```diff
@@ -3,7 +3,9 @@
 class ClassificationTest < ActiveSupport::TestCase
 
   def setup
+    hostgroup = FactoryGirl.create(:hostgroup)
```

Comment:
This is creating a new hostgroup in every test, can you move it to only the relevant test (I think it should be only the  one introduced in this PR) and update the host with that host group there?